src/ows/src/main/java/org/geoserver/ows/Dispatcher.java

```diff
@@ -1151,6 +1151,10 @@ Service findService(String id, String ver, String namespace) throws ServiceExcep
         if (matches.size() > 1) {
             List vmatches = new ArrayList(matches);
 
+            if (isCiteCompliant() && version == null && "WFS".equals(id)) {
```

Comment:
I agree with @aaime. The way I see it since the dispatcher needs to be aware of CITE compliance we need to make that awarenesses more configurable than a simple flag.